src/ProtectorServiceProvider.php

```diff
@@ -41,13 +42,18 @@ public function boot(): void
      */
     public function register(): void
     {
-        // Automatically apply the package configuration
+        // Automatically apply the package configuration.
         $this->mergeConfigFrom(__DIR__ . '/../config/protector.php', 'protector');
 
-        // Register the main class to use with the facade
+        // Register the main class to use with the facade.
         $this->app->singleton('protector', function () {
             return new Protector;
         });
+
+        // Register the SchemaState proxy classes.
+        $this->app->bind(MySqlSchemaStateProxy::class, function ($app, array $params) {
```

Comment:
if this approach is supposed to be flexible or modular, wouldnt it make sense to have this configurable and loaded dynamically?

Reply:
Yes, that's the idea. This is a POC that only implements MySql.
The goal would be to be able to bind a custom SchemaStateProxy to any SchemaState, including SchemaStates that might stem from third-party packages adding support for databases Laravel doesn't support.

Reply:
fair nuff